src/test/java/hudson/maven/AbstractMavenTestCase.java

```diff
@@ -37,9 +40,15 @@ protected void setUp() throws Exception {
     @Override
     public CommandLauncher createComputerLauncher(EnvVars env) throws URISyntaxException, MalformedURLException {
         int sz = this.jenkins.getNodes().size();
+        final URL slaveJarURL;
+        try {
+            slaveJarURL = this.jenkins.getJnlpJars("slave.jar").getURL();
+        } catch (IOException ex) {
+            throw new AssertionError("Cannot retrieve slave.jar for the test", ex);
+        }
         return new CommandLauncher(String.format("\"%s/bin/java\" %s %s -jar \"%s\"",
                 new Object[]{System.getProperty("java.home"), JAVA_HEADLESS_OPT, SLAVE_DEBUG_PORT > 0 ?
                         " -Xdebug -Xrunjdwp:transport=dt_socket,server=y,address=" + (SLAVE_DEBUG_PORT + sz) : "",
-                        (new File(this.jenkins.getJnlpJars("slave.jar").getURL().toURI())).getAbsolutePath()}), env);
+                        (new File(slaveJarURL.toURI())).getAbsolutePath()}), env);
```

Comment:
Can you just use `JenkinsRule`?

Reply:
I could, but I do not want to spend too much time on refactoring this plugin